### ISSUE TITLE
Just to check out: What is the basic component in GR-KAN, compared to nn.Linear in MLP?

### ISSUE THREAD

**Issue by lgy112112** [Adamdad/kat#19]
## Issue with Basic Component Structure in `KAN` Class

Firstly, I'd like to express my appreciation for your innovative work and the ideas presented in both your codebase and the accompanying documentation. It's truly inspiring!

### Question on `KAN` Class Structure in `katransformer.py`

In the file `kat/katransformer.py`, at line 103, I noticed the structure of the `KAN` class is defined as follows:

```python
class KAN(nn.Module):
    def __init__(
            self,
            in_features,
            hidden_features=None,
            out_features=None,
            act_layer=KAT_Group,
            norm_layer=None,
            bias=True,
            drop=0.,
            use_conv=False,
            act_init="gelu",
    ):
        super().__init__()
        out_features = out_features or in_features
        hidden_features = hidden_features or in_features
        bias = to_2tuple(bias)
        drop_probs = to_2tuple(drop)
        linear_layer = partial(nn.Conv2d, kernel_size=1) if use_conv else nn.Linear

        self.fc1 = linear_layer(in_features, hidden_features, bias=bias[0])
        self.act1 = KAT_Group(mode="identity")
        self.drop1 = nn.Dropout(drop_probs[0])
        self.norm = norm_layer(hidden_features) if norm_layer is not None else nn.Identity()
        self.act2 = KAT_Group(mode=act_init)
        self.fc2 = linear_layer(hidden_features, out_features, bias=bias[1])
        self.drop2 = nn.Dropout(drop_probs[1])

    def forward(self, x):
        x = self.act1(x)
        x = self.drop1(x)
        x = self.fc1(x)
        x = self.act2(x)
        x = self.drop2(x)
        x = self.fc2(x)
        return x
```

### Clarification Needed on the Basic Component of `GR-KAN`

Based on this setup, I assume the basic computational block of `GR-KAN` might be defined as the combination of:
```python
x = self.act1(x)
x = self.drop1(x)
x = self.fc1(x)
```

However, I would appreciate it if you could clarify whether this assumption is correct. I see significant customization in `KAN`, which makes it difficult to determine the precise role of each component.

### Question on Sequence: `act -> drop -> fc`

Could you also explain why the sequence is set as `activation -> dropout -> fully connected layer`? My intuition would suggest placing the `fc` layer first, followed by activation and dropout. I’m interested in understanding the rationale behind this particular ordering.

Thank you for your time and assistance in addressing these questions.

**Comment by lgy112112** [comment]
Nice. You gave the insight.

Still I am confused: if I take nn.Linear is the basic of mlp, KANLinear is the basic of KAN, then what is the basic of GR-KAN?

**Comment by lgy112112** [comment]
Nice @Adamdad . You gave the insight.

Still I am confused: if I take nn.Linear is the basic of mlp, KANLinear is the basic of KAN, then what is the basic of GR-KAN?

**Comment by Adamdad** [comment]
@lgy112112 , It should be a `KAT_Group -> Linear` combined layer. `KAT_Group` adds basis for each channel and `Linear` mixing their informations. Both are essential.
